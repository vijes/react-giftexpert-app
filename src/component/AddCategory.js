import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {

    const [inputvalue, setinputvalue] = useState('')

    const handleInputChange = ( e ) => {
        // console.log(e.target.value);
        setinputvalue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if ( inputvalue ) {
            setCategories( cats => [ inputvalue, ...cats ]);
            setinputvalue('');
        }else{
            console.log('No debe ser vacio el input');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p> { inputvalue } </p>
            <input 
                type="text"
                value={ inputvalue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}