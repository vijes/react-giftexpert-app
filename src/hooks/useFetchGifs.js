import { useEffect, useState } from 'react';
import { getGifs } from "../helpers/getGifs";


export const useFetchGifs = ( category )=> {
    
    const [state, setState] = useState({
        data: [], 
        loading: true
    });

    useEffect( () => {
        getGifs( category ).then(imgsResponse => {
            setState({
                data: imgsResponse,
                loading: false
            });
        });
    }, [ category ])

    return state; // data del arreglo declarado de un objeto
}