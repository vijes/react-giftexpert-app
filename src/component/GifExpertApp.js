import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['One punch', 'Samurai X', 'Dragon Ball'];

    // const [categories, setCategories] = useState(['Samurai X']);
    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {
    //     setCategories( ['Dragon Ball GT', ...categories] );
    //     // setCategories( cats => [ ... cats, 'Dagon Ball GT']);
    // }

    return (
        <>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category } 
                        />
                    ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp;