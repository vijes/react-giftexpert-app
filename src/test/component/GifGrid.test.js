import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../component/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas para componente <GifGrid />', () => {

    const category = 'Dragon Ball Z';
    
    test('Se debe consumir el componente y generar el snapshot.', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={ category }/> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes con useFetfchGifs()', () => {

        const gifs = [{
            id: 'abc',
            url: 'https://localhost:/imag',
            title: 'Imagen prueba'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow( <GifGrid category={ category }/> );

        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
        
    });
    
    
    
});