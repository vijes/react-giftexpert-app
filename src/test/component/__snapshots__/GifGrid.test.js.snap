// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para componente <GifGrid /> Se debe consumir el componente y generar el snapshot. 1`] = `
<Fragment>
  <h3>
    Dragon Ball Z
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
